extension/encoding/awslogsencodingextension/internal/unmarshaler/vpc-flow-log/vpc_flow_log_unmarshaler.go

```diff
@@ -0,0 +1,396 @@
+	// log record to add to the scope of logs
+	// of the resource
+	key := &resourceKey{}
+	record := plog.NewLogRecord()
+
+	// There are 4 fields for the addresses: srcaddr, pkt-srcaddr,
+	// dstaddr, pkt-dstaddr. We will save these fields in a
+	// map, so we can use the right conventions in the end.
+	//
+	// See https://docs.aws.amazon.com/vpc/latest/userguide/flow-logs-records-examples.html#flow-log-example-nat.
+	ips := make(map[string]string, 4)
+
+	// range over the fields of the log line
+	for i, field := range fields {
+		value := values[i]
+		if value == "-" {
+			// If a field is not applicable or could not be computed for a
+			// specific record, the record displays a '-' symbol for that entry.
+			//
+			// See https://docs.aws.amazon.com/vpc/latest/userguide/flow-log-records.html.
+			continue
+		}
+
+		if strings.HasPrefix(field, "ecs-") {
+			v.logger.Warn("currently there is no support for ECS fields")
+			continue
+		}
+
+		found, err := handleField(field, value, record, ips, key)
+		if err != nil {
+			return err
+		}
+		if !found {
+			v.logger.Warn("field is not an available field for a flow log record",
+				zap.String("field", field),
+				zap.String("documentation", "https://docs.aws.amazon.com/vpc/latest/userguide/flow-log-records.html"),
+			)
+		}
+	}
+
+	// get the address fields
+	addresses := handleAddresses(ips)
+	for field, value := range addresses {
+		record.Attributes().PutStr(field, value)
+	}
+
+	scopeLogs := v.getScopeLogs(*key, scopeLogsByResource)
+	rScope := scopeLogs.LogRecords().AppendEmpty()
+	record.MoveTo(rScope)
+
+	return nil
+}
+
+// handleAddresses creates a new map where the original field
+// names will be the known conventions for the fields
+func handleAddresses(addresses map[string]string) map[string]string {
+	// max is 3 fields, see example in
+	// https://docs.aws.amazon.com/vpc/latest/userguide/flow-logs-records-examples.html#flow-log-example-nat
+	recordAttr := make(map[string]string, 3)
+	srcaddr, foundSrc := addresses["srcaddr"]
+	pktSrcaddr, foundSrcPkt := addresses["pkt-srcaddr"]
+	if !foundSrcPkt && foundSrc {
+		// there is no middle layer, and "srcaddr" field
+		// corresponds to the original source address.
+		recordAttr[conventions.AttributeSourceAddress] = srcaddr
+	} else if foundSrcPkt && foundSrc {
+		recordAttr[conventions.AttributeSourceAddress] = pktSrcaddr
+		if srcaddr != pktSrcaddr {
+			// srcaddr is the middle layer
+			recordAttr[conventions.AttributeNetworkPeerAddress] = srcaddr
+		}
+	}
+
+	dstaddr, foundDst := addresses["dstaddr"]
+	pktDstaddr, foundDstPkt := addresses["pkt-dstaddr"]
+	if !foundDstPkt && foundDst {
+		// there is no middle layer, and "dstaddr" field
+		// corresponds to the original destination address.
+		recordAttr[conventions.AttributeDestinationAddress] = dstaddr
+	} else if foundDstPkt && foundDst {
+		recordAttr[conventions.AttributeDestinationAddress] = pktDstaddr
+		if pktDstaddr != dstaddr {
+			// dstaddr is the middle layer
```

Comment:
Perhaps we should add a check here to see if `pktSrcaddr != srcaddr`, and log a warning in that case, as we will be discarding some data. Then we'll know if our assumption that only one of them differs is faulty.

Reply:
I have added a condition to see if we are overwritting the data and print a warning in that case.

Reply:
Perfect, thank you!